q2_types/feature_data/_format.py

```diff
@@ -185,10 +187,38 @@ def sniff(self):
     AlignedDNAFASTAFormat)
 
 
+class DifferentialFormat(model.TextFileFormat):
+    def validate(self, *args):
+        try:
+            md = qiime2.Metadata.load(str(self))
+        except qiime2.metadata.MetadataFileError as md_exc:
+            raise ValidationError(md_exc) from md_exc
+
+        if md.column_count == 0:
+            raise ValidationError(
+                    ('Differential format must contain'
+                     'at least 1 column')
+            )
+
+        md = md.to_dataframe()
```

Comment:
Can you get away with replacing L203-L210 with [`Metadata.filter_columns(column_type='numeric')`](https://dev.qiime2.org/latest/api-reference/metadata/#qiime2.metadata.Metadata.filter_columns), and just checking that the column count is the same before and after?

```python
filtered_md = md.filter_columns(column_type='numeric')
if filtered_md.column_count != md.column_count:
    raise ValidationError('Differential types...')
```

If so, you can add on some additional validation from that method (only if you want/need to).